cmd/main_test.go

```diff
@@ -301,3 +315,59 @@ var _ = DescribeTable("TestRunAccessCheck", func(user string, namespace string,
 		"patch",
 		false),
 )
+
+var _ = DescribeTable("TestGetUserNamespaces",
+	func(labelKey string, labelValues []string, expectedNamespaces []string) {
+		e := echo.New()
+
+		var req *labels.Requirement
+		var err error
+
+		// Create the label requirement based on the input
+		if len(labelValues) > 0 {
+			req, err = labels.NewRequirement(labelKey, selection.In, labelValues)
+		} else {
+			req, err = labels.NewRequirement(labelKey, selection.Exists, []string{})
+		}
+		Expect(err).NotTo(HaveOccurred(), "Error creating label requirement")
+
+		namespaces, err := getUserNamespaces(e, *req)
+		Expect(err).NotTo(HaveOccurred(), "Error getting user namespaces")
+
+		var actualNamespaces []string
+		for _, ns := range namespaces {
+			actualNamespaces = append(actualNamespaces, ns.Name)
```

Comment:
I think the entries will be more readable if we use an actual namespace object. it will also make the test more accurate, actually testing against the whole object returned. 
Take a look at Homaja's #40 , she created a fixture to help with that, we can use it here as well.

Reply:
I see that Homaja has updated her code again. I will work on adding beforeeach and aftereach.

Reply:
Please tag me for review when done, the test will change massively as a result so reviewing it now is a bit redundant